File: fix_app.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

const filePath = path.join(__dirname, 'src/App.vue');
let content = fs.readFileSync(filePath, 'utf-8');

// Find the location where we need to insert the code
const targetString = "// NOU: Funcție care salvează și închide automat popup-ul";
const provideCode = `provide('t', t)

// --- WATCH PE USER ---
watch(user, (newUser, oldUser) => {
  if (!newUser) {
    budgets.value = []
    alertedBudgets.value.clear()
  } else if (oldUser && newUser.uid !== oldUser.uid) {
    budgets.value = []
    alertedBudgets.value.clear()
    loadBudgets()
  } else if (!oldUser && newUser) {
    loadBudgets()
  }
})

`;

if (content.includes(targetString)) {
  // Remove any duplicate provide() or watch() that might be there
  content = content.replace(/provide\('t', t\)\s*\/\/ --- WATCH PE USER ---[\s\S]*?\}\)\)\s*\/\//g, '//');
  content = content.replace(/provide\('t', t\)\s*\/\//g, '//');
  
  // Insert the code before "NOU: Funcție"
  content = content.replace(
    targetString,
    provideCode + targetString
  );
}

fs.writeFileSync(filePath, content, 'utf-8');
console.log('✅ App.vue fixed: provide() and watch() added correctly');

File: fix_app.py
import re

with open('src/App.vue', 'r', encoding='utf-8') as f:
    content = f.read()

# Remove the literal \n that were added by regex
content = content.replace(r'\n\n// --- WATCH PE USER ---\nwatch(user, (newUser, oldUser) => {\n  if (!newUser) {\n    budgets.value = []\n    alertedBudgets.value.clear()\n  } else if (oldUser && newUser.uid !== oldUser.uid) {\n    budgets.value = []\n    alertedBudgets.value.clear()\n    loadBudgets()\n  } else if (!oldUser && newUser) {\n    loadBudgets()\n  }\n})'.replace('\\n', '\n'), '')

# Now add it correctly
watch_code = '''

// --- WATCH PE USER ---
watch(user, (newUser, oldUser) => {
  if (!newUser) {
    budgets.value = []
    alertedBudgets.value.clear()
  } else if (oldUser && newUser.uid !== oldUser.uid) {
    budgets.value = []
    alertedBudgets.value.clear()
    loadBudgets()
  } else if (!oldUser && newUser) {
    loadBudgets()
  }
})'''

content = re.sub(r"(provide\('t', t\))", r'\1' + watch_code, content)

with open('src/App.vue', 'w', encoding='utf-8') as f:
    f.write(content)

print('Watch added successfully')
